### PR TITLE
Bugfix/length error

### PR DESCRIPTION
**Pull request by ahaffemayer**

### CODE REVIEW

File: src/crocoddyl_motion_server_node.cpp [hunk 1]
```diff
@@ -56,19 +57,22 @@ namespace panda_torque_mpc
                           std::string ocp_solve_time_topic_pub
                           )
         {
+            std::cout << "not right: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 2]
```diff
@@ -56,19 +57,22 @@ namespace panda_torque_mpc
...
             bool params_success = true;
             ///////////////////
             // Load parameters
             ///////////////////
             std::string arm_id;
             params_success = get_param_error_tpl<std::string>(nh, arm_id, "arm_id") && params_success;
+            std::cout << "not right1: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: include/panda_torque_mpc/common.h [hunk 3]
```diff
@@ -51,6 +52,14 @@ namespace panda_torque_mpc {
         return pinocchio::buildReducedModel(model_pin_full, locked_joints_id, q0_full);
     }
 
+    inline boost::shared_ptr<pinocchio::GeometryModel> loadPandaGeometryModel(pinocchio::Model model_pin)
+    {
+        std::string urdf_path = ros::package::getPath("panda_torque_mpc") + "/urdf/robot.urdf";
+        auto collision_model = boost::make_shared<pinocchio::GeometryModel>();
+        pinocchio::urdf::buildGeom(model_pin, urdf_path, pinocchio::COLLISION, *collision_model);
+
```
Comment (Kotochleb) [inline include/panda_torque_mpc/common.h]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 4]
```diff
@@ -56,19 +57,22 @@ namespace panda_torque_mpc
...
 
             // Croco params
             int nb_shooting_nodes, nb_iterations_max, max_qp_iter;
             double dt_ocp,solver_termination_tolerance,qp_termination_tol_abs , qp_termination_tol_rel, w_frame_running, w_frame_terminal, w_frame_vel_running, w_frame_vel_terminal, w_x_reg_running, w_x_reg_terminal, w_u_reg_running;
             std::vector<double> diag_frame_vel, diag_q_reg_running, diag_v_reg_running, diag_u_reg_running, armature;
             std::vector<double> pose_e_c, pose_c_o_ref;  // px,py,pz, qx,qy,qz,qw
             bool reference_is_placement;
+            std::cout << "not right2: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 5]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
             params_success = get_param_error_tpl<std::string>(nh, ee_frame_name_, "ee_frame_name") && params_success;
 
             
+            std::cout << "not right3: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 6]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
...
 
             if (!params_success)
             {
                 throw std::invalid_argument("CrocoMotionServer: check the your ROS parameters");
             }
-            
+            std::cout << "not right4: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 7]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
...
+
             model_pin_ = loadPandaPinocchio();
             data_pin_ = pin::Data(model_pin_);
+            std::cout << "not right5: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 8]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
...
 
             // Creating the collision model
             std::string urdf_path = ros::package::getPath("panda_torque_mpc") + "/urdf/robot.urdf";
+            std::cout << "not right61: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 9]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
...
 
             // Building the GeometryModel
-            auto collision_model = boost::make_shared<pinocchio::GeometryModel>();
-            pinocchio::urdf::buildGeom(model_pin_, urdf_path, pinocchio::COLLISION, *collision_model);
+            boost::shared_ptr<pinocchio::GeometryModel> collision_model = boost::make_shared<pinocchio::GeometryModel>();
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
            auto collision_model = boost::make_shared<pinocchio::GeometryModel>();
```

File: src/crocoddyl_motion_server_node.cpp [hunk 10]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
...
+            collision_model = loadPandaGeometryModel(model_pin_);
+
             double radius = 0.35/2.0;
+            std::cout << "not right64: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 11]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
...
+
             auto geometry = pinocchio::GeometryObject::CollisionGeometryPtr(new hpp::fcl::Sphere(radius));
+            std::cout << "not right7: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 12]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
...
 
             pinocchio::SE3 obstacle_pose(Eigen::Quaterniond (1.,0.,0.,0.), Eigen::Vector3d (0,0,0.825));
             // pinocchio::SE3 obstacle_pose;
             // obstacle_pose.setIdentity();
             // obstacle_pose.trans << 0., 0., 0.;
+            std::cout << "not right8: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 13]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
...
 
             pinocchio::GeometryObject obstacle("obstacle", 0,0, geometry, obstacle_pose);
             collision_model->addGeometryObject(obstacle);
+            std::cout << "not right9: " << std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 14]
```diff
@@ -112,37 +116,49 @@ namespace panda_torque_mpc
...
 
 
             assertm(collision_model->getGeometryId("obstacle") < collision_model->geometryObjects.size(), "The index of the obstacle is not right.");
             assertm(collision_model->getGeometryId("panda_leftfinger_0") < collision_model->geometryObjects.size(), "The index of the panda_leftfinger_0 is not right.");
             assertm(collision_model->getGeometryId("panda_rightfinger_0") < collision_model->geometryObjects.size(), "The index of the panda_rightfinger_0 is not right.");
 
+            std::cout << "not right10: " << std::endl;
+
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 15]
```diff
@@ -572,11 +588,17 @@ namespace panda_torque_mpc
 int main(int argc, char **argv)
 {
 
+    std::cout << "OK1: "<< std::endl;
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 16]
```diff
@@ -572,11 +588,17 @@ namespace panda_torque_mpc
...
     ros::init(argc, argv, "crocoddyl_motion_server_node");
+    std::cout << "OK2: "<< std::endl;
+
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 17]
```diff
@@ -572,11 +588,17 @@ namespace panda_torque_mpc
...
     ros::NodeHandle nh;
+    std::cout << "OK3: "<< std::endl;
+
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 18]
```diff
@@ -572,11 +588,17 @@ namespace panda_torque_mpc
...
     std::string robot_sensors_topic_sub = "robot_sensors";
     std::string control_topic_pub = "motion_server_control";
-    
+    std::cout << "OK4: "<< std::endl;
+
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 19]
```diff
@@ -586,6 +608,8 @@ int main(int argc, char **argv)
     std::string cam_pose_error_topic_pub = "cam_pose_error";
     std::string ee_pose_error_topic_pub = "ee_pose_error";
     std::string ocp_solve_time_topic_pub = "ocp_solve_time";
+    std::cout << "OK5: "<< std::endl;
+
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: src/crocoddyl_motion_server_node.cpp [hunk 20]
```diff
@@ -600,6 +624,8 @@ int main(int argc, char **argv)
                             ee_pose_error_topic_pub,
                             ocp_solve_time_topic_pub
                             );
+    std::cout << "OK6: "<< std::endl;
+
```
Comment (Kotochleb) [inline src/crocoddyl_motion_server_node.cpp]:
```suggestion
```

File: include/panda_torque_mpc/common.h [hunk 21]
```diff
@@ -51,6 +52,14 @@ namespace panda_torque_mpc {
         return pinocchio::buildReducedModel(model_pin_full, locked_joints_id, q0_full);
     }
 
+    inline boost::shared_ptr<pinocchio::GeometryModel> loadPandaGeometryModel(pinocchio::Model model_pin)
+    {
+        std::string urdf_path = ros::package::getPath("panda_torque_mpc") + "/urdf/robot.urdf";
```
Comment (Kotochleb) [inline include/panda_torque_mpc/common.h]:
```suggestion
        const std::string urdf_path = ros::package::getPath("panda_torque_mpc") + "/urdf/robot.urdf";
```
The value of this variable is not supposed to change later on, so it is better to keep it as a constant

File: include/panda_torque_mpc/common.h [hunk 22]
```diff
@@ -51,6 +52,14 @@ namespace panda_torque_mpc {
         return pinocchio::buildReducedModel(model_pin_full, locked_joints_id, q0_full);
     }
 
+    inline boost::shared_ptr<pinocchio::GeometryModel> loadPandaGeometryModel(pinocchio::Model model_pin)
```
Comment (Kotochleb) [inline include/panda_torque_mpc/common.h]:
```suggestion
    inline boost::shared_ptr<pinocchio::GeometryModel> loadPandaGeometryModel(const pinocchio::Model& model_pin)
```
Non-mutable parameters should always be constant. Also, anything that is not a fundamental type should be passed by refference

**Review by Kotochleb — CHANGES_REQUESTED** [review]
Please remove the unnecessary prints you used for debug

**Comment by ahaffemayer** [comment]
ready to merge @Kotochleb , it works after building ir.

**Review by Kotochleb — APPROVED** [review]

**Pull request merged**